src/utils/tools/updatingIcons/prepareTheUpdate.ts

```diff
@@ -0,0 +1,28 @@
+import { IconInterface } from 'components/icons'
+
+const prepareUpdatedIcon = (newIconDetails: IconInterface) => {
+  const updateDetails = {
+    $set: {},
+    $addToSet: {}
+  }
+  // loop through the keys
+  for (const [key, value] of Object.entries(newIconDetails)) {
+    if (key !== 'name' && key !== 'tags') {
+      updateDetails.$set[key] = value
+    }
+    if (key === 'tags') {
+      updateDetails.$addToSet[key] = { $each: value }
+    }
+    // category might be a string or an array
```

Comment:
Category is sometimes string, and sometimes an array, this check can be omitted, if `eos-icons.json` used a fixed schema for all the icons.